helix-core/src/main/java/org/apache/helix/controller/rebalancer/topology/Topology.java

```diff
@@ -230,16 +220,57 @@ private boolean isInstanceEnabled(ClusterConfig clusterConfig, String instanceNa
         || !clusterConfig.getDisabledInstances().containsKey(instanceName)));
   }
 
+  /**
+   * Populate clusterTopologyKeys and defaultDomainPathValues from clusterConfig
+   *
+   * @param clusterTopologyKeys       out parameter. LinkedHashSet to be populated for cluster
+   *                                  topology keys. The set will remain empty if topology aware is
+   *                                  not enabled or this cluster uses zone instead of domains.
+   * @param defaultDomainPathValues   out parameter. Map to be populated for all default path keys.
+   *                                  The map will remain empty if topology aware is not enabled or
+   *                                  this cluster uses zone instead of domains.
+   * @return lastValidType in clusterConfig.topology
+   */
+  private static String populateClusterTopologySetting(ClusterConfig clusterConfig,
+      LinkedHashSet<String> clusterTopologyKeys /*OUT*/,
+      Map<String, String> defaultDomainPathValues /*OUT*/) {
```

Comment:
Using input parameters to populate output values is generally not a best practice, maybe fine given this is just private method. But let us see if we find a better way to return these values.

Reply:
Updated.